native/src/main/java/io/ballerina/stdlib/io/nativeimpl/CsvChannelUtils.java

```diff
@@ -0,0 +1,171 @@
+                String fieldName = internalStructField.getFieldName();
+                if (fields.length > i) {
+                    String value = fields[i];
+                    switch (type) {
+                        case TypeTags.INT_TAG:
+                        case TypeTags.FLOAT_TAG:
+                        case TypeTags.STRING_TAG:
+                        case TypeTags.DECIMAL_TAG:
+                        case TypeTags.BOOLEAN_TAG:
+                            populateRecord(type, struct, fieldName, value);
+                            break;
+                        case TypeTags.UNION_TAG:
+                            List<Type> members = ((UnionType) internalStructField.getFieldType()).getMemberTypes();
+                            if (members.get(0).getTag() == TypeTags.NULL_TAG) {
+                                populateRecord(members.get(1).getTag(), struct, fieldName, value);
+                            } else if (members.get(1).getTag() == TypeTags.NULL_TAG) {
+                                populateRecord(members.get(0).getTag(), struct, fieldName, value);
+                            } else {
+                                throw IOUtils.createError("unsupported nillable field for value: " + value);
+                            }
+                            break;
+                        default:
+                            throw IOUtils.createError(
+                                    "type casting support only for int, float, boolean and string. "
+                                            + "Invalid value for the struct field: " + value);
+                    }
+                } else {
+                    struct.put(fieldName, null);
+                }
+            }
+        }
+        return struct;
+    }
+
+    private static void populateRecord(int type, Map<String, Object> struct, String fieldName, String value) {
+        if (value == null || value.isEmpty()) {
+            struct.put(fieldName, null);
+        } else {
+            String trimmedValue = value.trim();
+            switch (type) {
+                case TypeTags.INT_TAG:
+                    struct.put(fieldName, Long.parseLong(trimmedValue));
+                    return;
+                case TypeTags.FLOAT_TAG:
+                    struct.put(fieldName, Double.parseDouble(trimmedValue));
+                    break;
+                case TypeTags.DECIMAL_TAG:
+                    struct.put(fieldName, ValueCreator.createDecimalValue(trimmedValue));
+                    break;
+                case TypeTags.STRING_TAG:
+                    struct.put(fieldName, trimmedValue);
+                    break;
+                case TypeTags.BOOLEAN_TAG:
+                    struct.put(fieldName, Boolean.parseBoolean(trimmedValue));
+                    break;
+                default:
+                    throw IOUtils.createError("type casting support only for int, float, boolean and string. "
```

Comment:
Should improve the error message. Here it is not invalid value. Its unsupported type.